lib/note-content-editor.tsx

```diff
@@ -425,6 +426,44 @@ class NoteContentEditor extends Component<Props> {
     this.editor = editor;
     this.monaco = monaco;
 
+    monaco.languages.registerLinkProvider('plaintext', {
+      provideLinks: (model) => {
+        const matches = model.findMatches(
+          'simplenote://note/[a-zA-Z0-9-]+',
+          true, // searchOnlyEditableRange
+          true, // isRegex
+          false, // matchCase
+          null, // wordSeparators
+          false // captureMatches
+        );
+        const links = [];
+        matches.forEach(function (match) {
+          links.push({
+            range: match.range,
+            // don't set URL here because then Monaco skips resolveLink
+            // @cite: https://github.com/Microsoft/vscode/blob/8f89095aa6097f6e0014f2d459ef37820983ae55/src/vs/editor/contrib/links/getLinks.ts#L43:L65
+          });
+        });
+        return {
+          links: links,
+        };
```

Comment:
in some performance-critical code we can use a `for` loop to speed up iteration in JS but `.forEach` runs comparable to `.map` and in this case we're giving up semantic clarify by looping and mutating. how about a one-step transform?

```ts
return {
	// don't set URL here because then Monaco skips resolveLink
	// @cite: https://github.com/Microsoft/vscode/blob/8f89095aa6097f6e0014f2d459ef37820983ae55/src/vs/editor/contrib/links/getLinks.ts#L43:L65
	links: matches.map( ( { range } ) => ( { range } ) )
}
```